README.md

```diff
@@ -109,19 +140,15 @@ Whereas a subcircuit under that root may progress through these:
 ]
 ```
 
-Under this example, using product order, `(0, 0) <= (0, 1) <= (1, 1)`, but neither `(0, 2) <= (1, 1)` or `(1, 1) <= (2, 0)`. In all of these cases, the epoch of each timestamp is the first component of the pair, and the iteration is given by its second.
-
 ## 2.6 Circuit
 
-A circuit is an embedding of a PomoRA query plan into a directed graph whose vertices, called [nodes](#28-node), represent computation against [streams](#27-stream), and whose edges describe those streams.
+A circuit is an embedding of a [PomoRA] query plan into a directed (potentially cyclic) graph whose vertices ([nodes]) represent computation over [streams], and whose edges describe those streams.
 
-Circuits may contain subcircuits, representing recursive subcomputations that evaluate to a fixed point every iteration.
-
-Each iteration, a circuit is evaluated by evaluating its nodes in topological order, until each has reached a fixed point.
+Circuits MAY contain subcircuits, which MUST represent recursive subcomputations that evaluate to a fixed point every iteration. For each iteration, a circuit is run by evaluating its nodes (subcircuits and single nodes) in some topologically sorted order, until a fixed point is reached.
 
 ## 2.7 Stream
 
-A stream is an infinite sequence of values, each associated with subsequent timestamps. It is RECOMMENDED that streams not be reified directly, and they MAY instead be modeled as a cell containing the singular value in the stream at the current timestamp.
```

Comment:
@QuinnWilton 

> each associated with subsequent timestamps

Can you expand on the "subsequent" part? They get tagged on each iteration they touch?

Reply:
I also guess that "singlular value" is the element field of the triple?

Reply:
> Can you expand on the "subsequent" part?

Each node processes a single tuple at a time, incrementing that node's clock after each tuple.

> I also guess that "singlular value" is the element field of the triple?

As in, the tuple being processed at that timestamp. For example, given:

```elixir
{{k0, v0}, t0, 1}
{{k1, v1}, t1, 1}
{{k2, v2}, t2, 1}
```

At any point, each node would be processing one of the complete `{e, t, w}` triples, with the node's timestamp being incremented after each triple.

Reply:
Thanks! How do you feel about this:

> A stream is an infinite sequence of values. A node processing a stream progresses one tuple at a time, and increments its counter after every tuple. It is RECOMMENDED that streams not be reified directly, and they MAY instead be modeled as cells containing the element field in the stream at the current timestamp.

Reply:
Looks great!